### DISCUSSION TITLE
What does mean [ERROR GPL-0305] RePlAce diverged at newStepLength.

### DISCUSSION THREAD

**Discussion by titan73** [The-OpenROAD-Project/OpenROAD#6338] [Q&A] [answered]
What does it mean?
It seems it happen when density is too high.

**Comment by QuantamHD** [comment, marked as answer]
In general it means that the system has too many constraints, and there's no solution to the gradient descent problem.

In practice it's often that the density or routing is too dense, and the global placer can't find a way to fit everything in. The error message could be improved. It's a signal by the tool that you need to reconstrain your design in some way.

**Reply by titan73** [reply]
Ok thanks for the explanation. Indeed the message could be improved. Someone else has just entered an issue https://github.com/The-OpenROAD-Project/OpenROAD/issues/6323.
